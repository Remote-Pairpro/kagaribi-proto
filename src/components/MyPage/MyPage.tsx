import React from "react";

interface IMyPageProps {
    authUser: firebase.User;
}
class MyPage extends React.Component<IMyPageProps> {
    constructor(myPageProps: IMyPageProps) {
        super(myPageProps);
    }

    public render() {
        return (
            <div>
                <p>email: {this.props.authUser.email}</p>
            </div>
        );
    }
}

export default MyPage;
