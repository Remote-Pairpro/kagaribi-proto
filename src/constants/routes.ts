export const HOME = "/";
export const ABOUT = "/about";
export const SIGN_UP = "/signup";
export const SIGN_IN = "/signin";
