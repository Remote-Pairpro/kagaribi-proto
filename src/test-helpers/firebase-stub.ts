const firebaseUserDummy: firebase.User = {
    delete: null,
    emailVerified: null,
    getIdTokenResult: null,
    getIdToken: null,
    isAnonymous: null,
    linkAndRetrieveDataWithCredential: null,
    linkWithCredential: null,
    linkWithPhoneNumber: null,
    linkWithPopup: null,
    linkWithRedirect: null,
    metadata: null,
    phoneNumber: null,
    providerData: null,
    reauthenticateAndRetrieveDataWithCredential: null,
    reauthenticateWithCredential: null,
    reauthenticateWithPhoneNumber: null,
    reauthenticateWithPopup: null,
    reauthenticateWithRedirect: null,
    refreshToken: null,
    reload: null,
    sendEmailVerification: null,
    toJSON: null,
    unlink: null,
    updateEmail: null,
    updatePassword: null,
    updatePhoneNumber: null,
    updateProfile: null,
    displayName: null,
    email: null,
    photoURL: null,
    providerId: null,
    uid: null,
};

export { firebaseUserDummy };
