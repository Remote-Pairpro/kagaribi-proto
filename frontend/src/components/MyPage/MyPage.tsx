import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemText from "@material-ui/core/ListItemText";
import EmailIcon from "@material-ui/icons/Email";
import PermIdentityIcon from "@material-ui/icons/PermIdentity";
import React from "react";
import { Redirect, Switch } from "react-router-dom";
import * as routes from "../../constants/routes";

interface IMyPageProps {
    authUser: firebase.User | null;
}
class MyPage extends React.Component<IMyPageProps> {
    constructor(myPageProps: IMyPageProps) {
        super(myPageProps);
    }

    public render() {
        const user = this.props.authUser;
        return user ? (
            <div>
                <h1>MyPage</h1>
                <List>
                    <ListItem>
                        <EmailIcon />
                        <ListItemText>{user.email}</ListItemText>
                    </ListItem>
                    <ListItem>
                        <PermIdentityIcon />
                        <ListItemText> {user.uid}</ListItemText>
                    </ListItem>
                </List>
            </div>
        ) : (
            <Switch>
                <Redirect from={routes.MY_PAGE} to={routes.HOME} />
            </Switch>
        );
    }
}

export default MyPage;
