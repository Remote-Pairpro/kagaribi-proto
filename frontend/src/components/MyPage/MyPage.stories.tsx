import { storiesOf } from "@storybook/react";
import React from "react";
import { MemoryRouter } from "react-router";
import * as routes from "../../constants/routes";
import { firebaseUserDummy } from "../../test-helpers/firebase-stub";
import MyPage from "./MyPage";

storiesOf("MyPage", module).add("Auth", () => {
    return <MyPage authUser={firebaseUserDummy} />;
});

storiesOf("MyPage", module)
    .addDecorator(story => (
        <MemoryRouter initialEntries={[routes.HOME]}>{story()}</MemoryRouter>
    ))
    .add("Non Auth", () => {
        return <MyPage authUser={null} />;
    });
