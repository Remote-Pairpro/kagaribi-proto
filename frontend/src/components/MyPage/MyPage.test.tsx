import EmailIcon from "@material-ui/icons/Email";
import PermIdentityIcon from "@material-ui/icons/PermIdentity";
import { render, shallow } from "enzyme";
import React from "react";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import * as routes from "../../constants/routes";
import { firebaseUserDummy } from "../../test-helpers/firebase-stub";
import MyPage from "./MyPage";

describe("<MyPage authUser={user} />", () => {
    let user: firebase.User;
    beforeEach(() => {
        user = firebaseUserDummy;
    });

    it("should have text 'Mypage'", () => {
        const wrapper = render(<MyPage authUser={user} />);
        expect(wrapper.text()).toContain("MyPage");
    });

    it("should have icons", () => {
        const wrapper = shallow(<MyPage authUser={user} />);
        expect(wrapper.find(EmailIcon).length).toBe(1);
        expect(wrapper.find(PermIdentityIcon).length).toBe(1);
    });

    it("should renders correctly", () => {
        const tree = renderer.create(<MyPage authUser={user} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("<MyPage authUser={null} />", () => {
    it("should renders correctly", () => {
        const tree = renderer
            .create(
                <MemoryRouter initialEntries={[routes.HOME]}>
                    <MyPage authUser={null} />
                </MemoryRouter>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
