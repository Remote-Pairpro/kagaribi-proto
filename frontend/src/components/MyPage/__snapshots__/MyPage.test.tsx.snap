// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyPage authUser={null} /> should renders correctly 1`] = `null`;

exports[`<MyPage authUser={user} /> should renders correctly 1`] = `
<div>
  <h1>
    MyPage
  </h1>
  <ul
    className="MuiList-root-1 MuiList-padding-2"
  >
    <li
      className="MuiListItem-root-5 MuiListItem-default-8 MuiListItem-gutters-12"
      disabled={false}
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-16"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
      <div
        className="MuiListItemText-root-25"
      >
        <span
          className="MuiTypography-root-31 MuiTypography-subheading-38 MuiListItemText-primary-28"
        >
          example@example.com
        </span>
      </div>
    </li>
    <li
      className="MuiListItem-root-5 MuiListItem-default-8 MuiListItem-gutters-12"
      disabled={false}
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-16"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 5.9c1.16 0 2.1.94 2.1 2.1s-.94 2.1-2.1 2.1S9.9 9.16 9.9 8s.94-2.1 2.1-2.1m0 9c2.97 0 6.1 1.46 6.1 2.1v1.1H5.9V17c0-.64 3.13-2.1 6.1-2.1M12 4C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 9c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
      <div
        className="MuiListItemText-root-25"
      >
        <span
          className="MuiTypography-root-31 MuiTypography-subheading-38 MuiListItemText-primary-28"
        >
           
          this_is_example
        </span>
      </div>
    </li>
  </ul>
</div>
`;
